src/test/kotlin/com/example/flashcardbackend/deck/DeckControllerTest.kt

```diff
@@ -0,0 +1,186 @@
+package com.example.flashcardbackend.deck
+
+import com.example.flashcardbackend.flashcard.FlashcardListItem
+import org.hamcrest.Matchers
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito.*
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.*
+
+@WebMvcTest(DeckController::class, DeckService::class)
+class DeckControllerTest(
+    @Autowired
+    private val mockMvc: MockMvc,
+) {
+
+    @MockBean
+    private lateinit var deckRepository: DeckRepository
+
```

Comment:
I would restructure this test class by writing nested tests (https://www.petrikainulainen.net/programming/testing/junit-5-tutorial-writing-nested-tests/). If done right, this approach will help you to configure the system under test in the setup methods which remove duplicate code and makes your tests easier to write and maintain.